antidox/tester.py

```diff
@@ -0,0 +1,79 @@
+""" Program that's gets the latest revisions and checks for PII and toxicity """
+import argparse
+import json
+import os
+import time
+import datetime
+import requests
+import pywikibot
+from antidox import clean
+from antidox import perspective
+
+
+def wiki_write(result, header):
+  """ Writes results to wikipedia/wikimedia bot page """
+  pywikibot.config.register_family_file('doxwiki', os.path.join(os.path.dirname(__file__), 'doxwiki_family.py'))
+  pywikibot.config.usernames['doxwiki']['en'] = u'Antidoxer'
+  site = pywikibot.Site()
+  repo = site.data_repository()
+  page = pywikibot.Page(site, u"User_talk:Antidoxer")
+
+  heading = (header)
+  content = (result)
+  message = "\n\n{}\n{} --~~~~".format(heading, content)
+  page.save(summary="Testing", watch=None, minor=False, botflag=True,
+            force=False, async=False, callback=None,
+            apply_cosmetic_changes=None, appendtext=message)
+
+def log_change():
+  """ gets latest revisions and cleans them """
+  while True:
+    start = datetime.datetime.now() - datetime.timedelta(minutes=2)
```

Comment:
There's a slight problem with this code: it requests everything in the last 2 minutes, and it sleeps for 2 minutes. But the code takes time to run (if there are a lot of changes, it may take a while to make the Perspective/DLP requests for all the changes), so more than 2 minutes will pass before the next time we make the request. This means it's likely we'll miss some changes, which is bad.

We can be more exact and avoid missing changes, though. Something like this should work:

```
def log_change():
  start = datetime.datetime.now() - datetime.timedelta(minutes=2)
  while True:
    end = datetime.datetime.now()
    page = [... use rcstart=start.isoformat() and rcend=end.isoformat() ...]
    get_page = requests.get(page)
    # The next time we run the loop, the start time will be this last request's end time.
    start = end  
    response = json.loads(get_page.content)
    [...rest of the code is the same...]
```

Reply:
This is a little confusinf to me, because the way I'm reading it, that's how it already is. You're saying to put the "rcend" in there? and set it equal to "now", but I already have that, the differen is I just manually put "now".

Reply:
It's different - I can explain in person.